docs/architecture/decisions/0005-admin-ui-adr.md

```diff
@@ -0,0 +1,91 @@
+# 5. Admin UI
+
+Date: 2025-09-17
+
+## Status
+
+Accepted
+
+## Context
+
+As a team building systems for the Reprocessors and Exporters side of the EPR (Extended Producer Responsibilities) service we need to fulfill the needs of various privileged stakeholders in the space including administrators of the service and the Regulator.
+
+At the moment, there isn't a way for those stakeholders to access the information submitted by the Operators (Reprocessors and Exporters) and kept in our service's database.
+
+It's especially pressing for the team supporting the EPR service to gain the ability to debug potential problems with the existing forms for submitting `organisation`, `registration` and `accreditation` information as soon as possible, as mentioned in the [Data Extraction ADR](0004-data-extraction.md).
+
+## Decision
+
+We have chosen to build an Admin UI (`epr-re-ex-admin-frontend`) for the EPR service.
+
+The goal of the service is to act as an administrative tool for both system administrators and for the Regulator of EPR service to perform their duties based on different user's authorisation levels.
+
+### Service features
+
+The service will have the following characteristics:
+
+- It will be based on the CDP Node.js Frontend Template
+- It will live in CDP's protected zone with all users signing in via AAD/SSO (Entra Id)
+- Authorisation will be based on role-based access control approach (RBAC) supporting only two different roles (at least initially): System Administrator and Regulator.
+- A simple secret-based mechanism will be used initially to map users to roles. In the future a more sophisticated mechanism that enables user management capabilities will need to be devised. This implies that user administration will be performed manually by the team in the initial phases.
+- Like on similar projects, all AAD (Entra Id) users will be allowed to login (authenticate). Access to the different resources will follow a "route guarding" approach by which different pages will require different user roles, i.e. a user without the right access level will receive 403 response and see a "You do not have access to this page" message when visiting that page/resource.
+- The app will get all its data from `epr-backend`. Calls to `epr-backend` will relay AAD's auth token to ensure the user is authorised to access the affected endpoint.
+- The app will not support multiple languages, at least not initially, and English will be the only language available.
+
+### Place in the Re-Ex EPR architecture
+
+The following diagram outlines how `epr-re-ex-admin-frontend` fits into the overall architecture of the EPR service:
+
+```mermaid
+flowchart TD;
```

Comment:
This needs updating to reflect our emerging understanding that

- `epr-frontend` will be deployed to the public zone, and be accessible by the general public (ie `User`)
- `epr-re-ex-admin-frontend` will be deployed to the public zone and
  - **not** be accessible by the general public (ie `User`)
  - accessible by `Service Maintainers` and `Regulators`
- both `frontend` services will interact with `epr-backend` in the protected zone

Reply:
I made a small change in the right direction.
Merging the PR now to close the ticket and I will discuss this again with you @crosslandwa when I move it to the new repo.